### PR TITLE
[CORE] Apply code-checks/format with misc-definitions-in-headers

### PR DESCRIPTION
**Pull request by smuzaffar**
code-checks/format applied on full cmssw

### CODE REVIEW

File: IOPool/Common/bin/EdmProvDump.cc [hunk 1]
```diff
@@ -1055,21 +1055,76 @@ int main(int argc, char* argv[]) {
...
+                                                                                             "ter "
+                                                                                             "set "
+                                                                                             "ID "
+                                                                                             "string"
+                                                                                             " (and "
+                                                                                             "print "
+                                                                                             "nothin"
+                                                                                             "g "
+                                                                                             "else"
+                                                                                             ")");
```
Comment (wddgit) [inline IOPool/Common/bin/EdmProvDump.cc:1127]: This looks worse than it was before. Several other of the modifications look worse, although this is the worst one  of all of them.

**Comment by makortel** [comment]
@cmsbuild please test

File: IOPool/Common/bin/EdmProvDump.cc (hunk 1 above)
Comment (smuzaffar) [reply IOPool/Common/bin/EdmProvDump.cc:1127]: wow, auto formating :-) yes this look bad

Comment (smuzaffar) [reply IOPool/Common/bin/EdmProvDump.cc:1127]: and few other places too :-(

File: FWCore/Integration/test/AcquireIntStreamFilter.cc [hunk 2]
```diff
@@ -33,7 +33,7 @@ namespace edmtest {
   };
 
   AcquireIntStreamFilter::AcquireIntStreamFilter(edm::ParameterSet const& pset)
-      : m_token{edm::Service<test_acquire::WaitingService>{}->getToken()} {
+      : m_token{edm::Service<test_acquire::WaitingService>{} -> getToken()} {
```
Comment (makortel) [inline FWCore/Integration/test/AcquireIntStreamFilter.cc:36]: This looks worse too (although only slightly compared to some of the others)

Comment (smuzaffar) [reply FWCore/Integration/test/AcquireIntStreamFilter.cc:36]: looks like a bug to me, we use google formating style and it [a] says 
```
No spaces around period or arrow. Pointer operators do not have trailing spaces.
```

and strangely clang-format only added spaces in this line while in the same file there are other instances of `->` where it did not suggest to add the spaces. May be it is becase of `{}` before the arrow. 
 
[a]
https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions

Comment (makortel) [reply FWCore/Integration/test/AcquireIntStreamFilter.cc:36]: > May be it is becase of `{}` before the arrow.

That hypothesis would be simple to test with
```suggestion
      : m_token{edm::Service<test_acquire::WaitingService>()->getToken()} {
```

Comment (smuzaffar) [reply FWCore/Integration/test/AcquireIntStreamFilter.cc:36]: yes, if we have `()->` then clang-format does not add extra spaces but for `{}->` it does. Should I apply your change @makortel ?

Comment (makortel) [reply FWCore/Integration/test/AcquireIntStreamFilter.cc:36]: Let's go for `()->` then.

**Comment by smuzaffar** [comment]
please test

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**